include/openenclave/internal/sgx/td.h

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_SGX_TD_H
+#define _OE_SGX_TD_H
+
+#include <openenclave/internal/defs.h>
+
+OE_EXTERNC_BEGIN
+
+/*
+**==============================================================================
+**
+** oe_thread_data_t
+**
+**     This structure defines information about an enclave thread. Each
+**     instance is associated with one thread control structure (TCS). This
+**     structure resides in the FS segment page (referenced by the FS segment
+**     register). A thread obtains its thread data structure by calling
+**     oe_get_thread_data(), which fetches the address at offset zero in
+**     the FS segment register (%fs:0) which contains
+**     oe_thread_data_t.self_addr.
+**
+**==============================================================================
+*/
+
+typedef struct _oe_thread_data oe_thread_data_t;
+
+/* Note: unused fields have a "__" prefix */
+struct _oe_thread_data
```

Comment:
Though this structure is named _oe_thread_data it, most if not all of its fields are mandated by the x64 Linux ABI.
That makes this an interesting structure; mostly ABI defined with couple of OE fields.